pkg/k8s/slim/k8s/api/core/v1/types.go

```diff
@@ -252,6 +252,11 @@ type PodSpec struct {
 	// More info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
 	// +optional
 	ServiceAccountName string `json:"serviceAccountName,omitempty" protobuf:"bytes,8,opt,name=serviceAccountName"`
+	// NodeName is a request to schedule this pod onto a specific node. If it is non-empty,
+	// the scheduler simply schedules this pod onto that node, assuming that it fits resource
+	// requirements.
+	// +optional
+	NodeName string `json:"nodeName,omitempty" protobuf:"bytes,10,opt,name=nodeName"`
```

Comment:
There's no need for this field actually because we know that any Pod object that enters the K8s watchers is a local pod due to https://github.com/cilium/cilium/blob/82c742f2e9fb65fb3fc392c7dcb3b4e22b69c650/pkg/k8s/watchers/pod.go#L122.

Reply:
The addition was based on a suggestion from @aanm in a discussion we had on Slack (with the relevant portion starting around [here](https://cilium.slack.com/archives/C53TG4J4R/p1645045322460259?thread_ts=1644882661.331769&cid=C53TG4J4R)). The idea was to play it safe since some code paths to get the pod seem to rely on [`GetCachedPod()`](https://github.com/cilium/cilium/blob/fcbeb6402b2ce63be373e27a216c4b6806188def/pkg/k8s/watchers/pod.go#L852), which comes with a comment saying that a `K8sEventHandover` determines whether the backing store is populated from local or cluster-wide pods. So the concern was that we could be seeing cluster-wide pods if the Cilium configuration was set accordingly.

With this additional context, can you confirm that omitting the `NodeName` addition (and consumption) is still the right thing to do?

Reply:
As discussed in Slack, we'll keep the nodeName addition for a future PR to fix an endpoint restoration bug (sep. from this issue). That issue is tracked in https://github.com/cilium/cilium/issues/18923.